### PR TITLE
Add configurable metrics interval

### PR DESCRIPTION
**Pull request by yvliu-amzn**
*Description of changes:*
This is to emit pipe mode metrics on a configurable interval.

*Testing:*
Unit testing;
Performance testing:
With 22G of data:
- Original code: epoch 1 - 207948ms, epoch 2 -  200785ms
- With new parameter disabled: epoch 1 - 211401ms, epoch 2 - 197084ms
- With new parameter enabled, interval = 10000 : epoch 1 - 209611ms, epoch 2 - 199258ms

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: create_integ_test_docker_images.py [hunk 1]
```diff
@@ -50,7 +50,7 @@
         cache_from=[tag],
         buildargs={'sagemaker_tensorflow': sdist_path,
                    'device': args.device,
-                   'python': 'python' if python_version == 2 else 'python3',
+                   'python': 'python3',
```
[the remark this answers is not in the document]
Comment (yvliu-amzn) [reply create_integ_test_docker_images.py]: 1. According to Nadia, VERSION file isn’t really used for pypi packages. What I need to do is to create PRs against both master and tf-2. 2. I'll include the change explanation in the commit message. 3. I have updated the path in the new revision.

File: src/pipemode_op/Dataset/src/pipemode_dataset_op_kernel.cpp [hunk 2]
```diff
@@ -184,8 +189,16 @@ class PipeModeDatasetOp : public DatasetOpKernel {
                         *end_of_sequence = true;
                     }
                     auto end = std::chrono::high_resolution_clock::now();
-                    read_time_ = read_time_ + std::chrono::duration_cast<std::chrono::nanoseconds>(end - start);
+                    auto delta_ns = std::chrono::duration_cast<std::chrono::nanoseconds>(end - start);
+                    read_time_ += delta_ns;
                     read_bytes_ += storage->size();
+                    num_samples_ ++;
```
[the remark this answers is not in the document]
Comment (yvliu-amzn) [reply src/pipemode_op/Dataset/src/pipemode_dataset_op_kernel.cpp]: Updated in the new revision.

File: src/pipemode_op/Dataset/src/pipemode_dataset_op_kernel.cpp [hunk 3]
```diff
@@ -194,11 +210,12 @@ class PipeModeDatasetOp : public DatasetOpKernel {
             ~Iterator() {
                 if (benchmark_) {
                     int64_t read_time_ms = std::chrono::duration_cast<std::chrono::milliseconds>(read_time_).count();
-                    std::cout << "PipeModeDatasetOp::Dataset::Iterator read_time_ms: " << read_time_ms << std::endl;
-                    std::cout << "PipeModeDatasetOp::Dataset::Iterator read_bytes: " << read_bytes_  << std::endl;
+                    std::cout << "PipeModeDatasetOp::Dataset::Iterator epoch read_time_ms: " << read_time_ms
```
Comment (owen-t) [inline src/pipemode_op/Dataset/src/pipemode_dataset_op_kernel.cpp]: Apologies for not picking this up in the first review. This needn't be an epoch. 

1) The algorithm that's acting on the data may just be a single pass, in which case there's no notion of an epoch.
2) The Dataset may be discarded before the entire fifo is drained.

I would replace epoch with 'total'.

File: src/sagemaker_tensorflow/pipemode.py [hunk 4]
```diff
@@ -53,6 +53,8 @@ def __init__(self, channel, record_format='RecordIO',
             state_dir: The directory where pipe index state is persisted.
             config_dir: The path for SageMaker input data config.
             benchmark: If True, causes the Dataset to emit timing and throughput metrics to stdout.
+            benchmark_records_interval: The number of records per interval to emit timing and throughput metrics
+                                        to stdout. If zero, no emitting the metrics.
```
Comment (owen-t) [inline src/sagemaker_tensorflow/pipemode.py]: This documentation is a little confusing - to the reader, it's not clear how ``benchmark`` and ``benchmark_records_interval`` interrelate.

benchmark: Controls whether to emit timing and throughput metrics after closing an Iterator created from this Dataset. If True, timing and throughput metrics will be emitted to stdout after an Iterator created from this Dataset is destroyed.
benchmark_records_interval: Controls whether to emit timing and throughput metrics while records are being read from this Dataset. Defines the number of records per interval to emit timing and throughput metrics. If zero, no metrics will be emitted while records are being read from this Dataset. Metrics are emitted to stdout.

File: test/unit/test_pipemode.py [hunk 5]
```diff
@@ -143,3 +143,25 @@ def test_multiple_iterators():
...
+def test_benchmark_records_interval_disabled(capfd):
+    channel, directory = write_to_channel("A", [b"bear"])
+
+    dataset = PipeModeDataset(channel, pipe_dir=directory, state_dir=directory, config_dir=directory, benchmark_records_interval=0)
+    with tf.Session() as sess:
+        it = dataset.make_one_shot_iterator()
+        next = it.get_next()
+        assert sess.run(next) == b"bear"
+        out, err = capfd.readouterr()
+        assert 'Iterator records' not in out
```
Comment (owen-t) [inline test/unit/test_pipemode.py]: Add a new line here

Comment (yvliu-amzn) [reply test/unit/test_pipemode.py]: fixed in the new revision.

File: src/sagemaker_tensorflow/pipemode.py (hunk 4 above)
Comment (yvliu-amzn) [reply src/sagemaker_tensorflow/pipemode.py]: fixed in the new revision.

File: src/pipemode_op/Dataset/src/pipemode_dataset_op_kernel.cpp (hunk 3 above)
Comment (yvliu-amzn) [reply src/pipemode_op/Dataset/src/pipemode_dataset_op_kernel.cpp]: fixed in the new revision.

**Pull request merged**